packages/astro/src/core/render-context.ts

```diff
@@ -591,8 +591,14 @@ export class RenderContext {
 				computedLocale = computeCurrentLocale(referer, locales, defaultLocale);
 			}
 		} else {
-			const pathname =
-				routeData.pathname && !isRoute404or500(routeData) ? routeData.pathname : url.pathname;
+			// For SSG we match the route naively, for dev we handle fallback on 404, for SSR we find route from fallbackRoutes
+			const route =
+				(routeData.pattern.test(url.pathname)
+					? routeData
+					: routeData.fallbackRoutes.find((fallbackRoute) =>
+							fallbackRoute.pattern.test(url.pathname),
+						)) ?? routeData;
```

Comment:
In the past we used these operators and they made the readability worse, and we decided to use simple if/else. We would appreciate it if we keep using them

Reply:
Sure thing. Will rewrite the logic here.